### PR TITLE
Allow for clean cache command to use config'd cache_dir (plus clean up)

### PR DESCRIPTION
**Pull request by isleshocky77**
## Description

* Update command to allow for cache directory specified in config
* Remove warnings/errors when no cached directory present
* Update to have namespace "cache" so we can add other cache commands later
  * Leave alias for current command
* Use provided toolsets required by robo for implementation
* Print cleaned directories
* Confirm before cleaning directory
  * Allow for force override

Related Issues / PRs: #6793 #6648 #6649 

## Motivation and Context

* Using a custom cache directory and the current implementation doesn't use the `$sugar_config['cache_dir']` variable.
* I would like to see what the command is going to do, before it runs
* I would like to be able to have other cache related commands (e.g.`cache:warm`)

## How To Test This

* Grab PR
* Change the `$sugar_config['cache_dir']` varible in `config_override.php`
* Run command `cache:clean` OR `clean:cache`
* Output should be like following

```
$ ./vendor/bin/robo clean:cache
➜  Found Sub-Cache Directories to Clean: 

 * var/cache/dashlets
 * var/cache/jsLanguage
 * var/cache/include/javascript
 * var/cache/themes
 * var/cache/Relationships

➜  Found Module-Cache Directories to Clean: 
 * var/cache/modules/EditCustomFields
 * var/cache/modules/Releases
 * var/cache/modules/Trackers
 * var/cache/modules/Alerts
 * var/cache/modules/SchedulersJobs
 * var/cache/modules/Calls
 * var/cache/modules/AOS_Invoices
 * var/cache/modules/AOP_Case_Events
 * var/cache/modules/Accounts
 * var/cache/modules/Roles
 * var/cache/modules/CampaignTrackers
 * var/cache/modules/Documents
 * var/cache/modules/SurveyQuestionOptions
 * var/cache/modules/Calendar
 * var/cache/modules/AOR_Conditions
 * var/cache/modules/Import_1
 * var/cache/modules/AOP_Case_Updates
 * var/cache/modules/Employees
 * var/cache/modules/Notes
 * var/cache/modules/Currencies
 * var/cache/modules/Spots
 * var/cache/modules/InboundEmail
 * var/cache/modules/jjwg_Maps
 * var/cache/modules/Project
 * var/cache/modules/AOR_Reports
 * var/cache/modules/EmailAddresses
 * var/cache/modules/Schedulers
 * var/cache/modules/DynamicFields
 * var/cache/modules/Bugs
 * var/cache/modules/Leads
 * var/cache/modules/SurveyQuestionResponses
 * var/cache/modules/AOBH_BusinessHours
 * var/cache/modules/AOD_IndexEvent
 * var/cache/modules/jjwg_Markers
 * var/cache/modules/ACL
 * var/cache/modules/OutboundEmailAccounts
 * var/cache/modules/Reminders_Invitees
 * var/cache/modules/AOR_Scheduled_Reports
 * var/cache/modules/EmailText
 * var/cache/modules/UserPreferences
 * var/cache/modules/Groups
 * var/cache/modules/Import_2
 * var/cache/modules/MergeRecords
 * var/cache/modules/DocumentRevisions
 * var/cache/modules/OAuth2Tokens
 * var/cache/modules/Prospects
 * var/cache/modules/ProspectLists
 * var/cache/modules/AOK_KnowledgeBase
 * var/cache/modules/AM_TaskTemplates
 * var/cache/modules/jjwg_Areas
 * var/cache/modules/AOS_Contracts
 * var/cache/modules/SavedSearch
 * var/cache/modules/FP_events
 * var/cache/modules/Favorites
 * var/cache/modules/ACLActions
 * var/cache/modules/Users
 * var/cache/modules/Campaigns
 * var/cache/modules/Relationships
 * var/cache/modules/AOW_WorkFlow
 * var/cache/modules/AOW_Actions
 * var/cache/modules/AOR_Fields
 * var/cache/modules/Opportunities
 * var/cache/modules/Tasks
 * var/cache/modules/EmailMarketing
 * var/cache/modules/EmailMan
 * var/cache/modules/Surveys
 * var/cache/modules/Administration
 * var/cache/modules/AOW_Processed
 * var/cache/modules/SecurityGroups
 * var/cache/modules/ACLRoles
 * var/cache/modules/Home
 * var/cache/modules/SugarFeed
 * var/cache/modules/SurveyResponses
 * var/cache/modules/Meetings
 * var/cache/modules/AOS_Products
 * var/cache/modules/CampaignLog
 * var/cache/modules/Connectors
 * var/cache/modules/ProjectTask
 * var/cache/modules/Import
 * var/cache/modules/AOK_Knowledge_Base_Categories
 * var/cache/modules/ModuleBuilder
 * var/cache/modules/TemplateSectionLine
 * var/cache/modules/Calls_Reschedule
 * var/cache/modules/Reminders
 * var/cache/modules/AM_ProjectTemplates
 * var/cache/modules/AOW_Conditions
 * var/cache/modules/OAuthKeys
 * var/cache/modules/jjwg_Address_Cache
 * var/cache/modules/Contacts
 * var/cache/modules/AOS_Line_Item_Groups
 * var/cache/modules/vCals
 * var/cache/modules/AOS_Product_Categories
 * var/cache/modules/AOS_PDF_Templates
 * var/cache/modules/OAuthTokens
 * var/cache/modules/Cases
 * var/cache/modules/AOS_Quotes
 * var/cache/modules/EAPM
 * var/cache/modules/Configurator
 * var/cache/modules/FP_Event_Locations
 * var/cache/modules/OAuth2Clients
 * var/cache/modules/AOD_Index
 * var/cache/modules/EmailTemplates
 * var/cache/modules/SurveyQuestions
 * var/cache/modules/AOS_Products_Quotes
 * var/cache/modules/AOR_Charts

 [Filesystem\CleanDir] Cleaned var/cache/dashlets
 [Filesystem\CleanDir] Cleaned var/cache/jsLanguage
 [Filesystem\CleanDir] Cleaned var/cache/include/javascript
 [Filesystem\CleanDir] Cleaned var/cache/themes
 [Filesystem\CleanDir] Cleaned var/cache/Relationships
 [Filesystem\CleanDir] Cleaned var/cache/modules/EditCustomFields
 [Filesystem\CleanDir] Cleaned var/cache/modules/Releases
 [Filesystem\CleanDir] Cleaned var/cache/modules/Trackers
 [Filesystem\CleanDir] Cleaned var/cache/modules/Alerts
 [Filesystem\CleanDir] Cleaned var/cache/modules/SchedulersJobs
 [Filesystem\CleanDir] Cleaned var/cache/modules/Calls
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Invoices
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOP_Case_Events
 [Filesystem\CleanDir] Cleaned var/cache/modules/Accounts
 [Filesystem\CleanDir] Cleaned var/cache/modules/Roles
 [Filesystem\CleanDir] Cleaned var/cache/modules/CampaignTrackers
 [Filesystem\CleanDir] Cleaned var/cache/modules/Documents
 [Filesystem\CleanDir] Cleaned var/cache/modules/SurveyQuestionOptions
 [Filesystem\CleanDir] Cleaned var/cache/modules/Calendar
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOR_Conditions
 [Filesystem\CleanDir] Cleaned var/cache/modules/Import_1
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOP_Case_Updates
 [Filesystem\CleanDir] Cleaned var/cache/modules/Employees
 [Filesystem\CleanDir] Cleaned var/cache/modules/Notes
 [Filesystem\CleanDir] Cleaned var/cache/modules/Currencies
 [Filesystem\CleanDir] Cleaned var/cache/modules/Spots
 [Filesystem\CleanDir] Cleaned var/cache/modules/InboundEmail
 [Filesystem\CleanDir] Cleaned var/cache/modules/jjwg_Maps
 [Filesystem\CleanDir] Cleaned var/cache/modules/Project
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOR_Reports
 [Filesystem\CleanDir] Cleaned var/cache/modules/EmailAddresses
 [Filesystem\CleanDir] Cleaned var/cache/modules/Schedulers
 [Filesystem\CleanDir] Cleaned var/cache/modules/DynamicFields
 [Filesystem\CleanDir] Cleaned var/cache/modules/Bugs
 [Filesystem\CleanDir] Cleaned var/cache/modules/Leads
 [Filesystem\CleanDir] Cleaned var/cache/modules/SurveyQuestionResponses
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOBH_BusinessHours
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOD_IndexEvent
 [Filesystem\CleanDir] Cleaned var/cache/modules/jjwg_Markers
 [Filesystem\CleanDir] Cleaned var/cache/modules/ACL
 [Filesystem\CleanDir] Cleaned var/cache/modules/OutboundEmailAccounts
 [Filesystem\CleanDir] Cleaned var/cache/modules/Reminders_Invitees
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOR_Scheduled_Reports
 [Filesystem\CleanDir] Cleaned var/cache/modules/EmailText
 [Filesystem\CleanDir] Cleaned var/cache/modules/UserPreferences
 [Filesystem\CleanDir] Cleaned var/cache/modules/Groups
 [Filesystem\CleanDir] Cleaned var/cache/modules/Import_2
 [Filesystem\CleanDir] Cleaned var/cache/modules/MergeRecords
 [Filesystem\CleanDir] Cleaned var/cache/modules/DocumentRevisions
 [Filesystem\CleanDir] Cleaned var/cache/modules/OAuth2Tokens
 [Filesystem\CleanDir] Cleaned var/cache/modules/Prospects
 [Filesystem\CleanDir] Cleaned var/cache/modules/ProspectLists
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOK_KnowledgeBase
 [Filesystem\CleanDir] Cleaned var/cache/modules/AM_TaskTemplates
 [Filesystem\CleanDir] Cleaned var/cache/modules/jjwg_Areas
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Contracts
 [Filesystem\CleanDir] Cleaned var/cache/modules/SavedSearch
 [Filesystem\CleanDir] Cleaned var/cache/modules/FP_events
 [Filesystem\CleanDir] Cleaned var/cache/modules/Favorites
 [Filesystem\CleanDir] Cleaned var/cache/modules/ACLActions
 [Filesystem\CleanDir] Cleaned var/cache/modules/Users
 [Filesystem\CleanDir] Cleaned var/cache/modules/Campaigns
 [Filesystem\CleanDir] Cleaned var/cache/modules/Relationships
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOW_WorkFlow
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOW_Actions
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOR_Fields
 [Filesystem\CleanDir] Cleaned var/cache/modules/Opportunities
 [Filesystem\CleanDir] Cleaned var/cache/modules/Tasks
 [Filesystem\CleanDir] Cleaned var/cache/modules/EmailMarketing
 [Filesystem\CleanDir] Cleaned var/cache/modules/EmailMan
 [Filesystem\CleanDir] Cleaned var/cache/modules/Surveys
 [Filesystem\CleanDir] Cleaned var/cache/modules/Administration
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOW_Processed
 [Filesystem\CleanDir] Cleaned var/cache/modules/SecurityGroups
 [Filesystem\CleanDir] Cleaned var/cache/modules/ACLRoles
 [Filesystem\CleanDir] Cleaned var/cache/modules/Home
 [Filesystem\CleanDir] Cleaned var/cache/modules/SugarFeed
 [Filesystem\CleanDir] Cleaned var/cache/modules/SurveyResponses
 [Filesystem\CleanDir] Cleaned var/cache/modules/Meetings
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Products
 [Filesystem\CleanDir] Cleaned var/cache/modules/CampaignLog
 [Filesystem\CleanDir] Cleaned var/cache/modules/Connectors
 [Filesystem\CleanDir] Cleaned var/cache/modules/ProjectTask
 [Filesystem\CleanDir] Cleaned var/cache/modules/Import
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOK_Knowledge_Base_Categories
 [Filesystem\CleanDir] Cleaned var/cache/modules/ModuleBuilder
 [Filesystem\CleanDir] Cleaned var/cache/modules/TemplateSectionLine
 [Filesystem\CleanDir] Cleaned var/cache/modules/Calls_Reschedule
 [Filesystem\CleanDir] Cleaned var/cache/modules/Reminders
 [Filesystem\CleanDir] Cleaned var/cache/modules/AM_ProjectTemplates
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOW_Conditions
 [Filesystem\CleanDir] Cleaned var/cache/modules/OAuthKeys
 [Filesystem\CleanDir] Cleaned var/cache/modules/jjwg_Address_Cache
 [Filesystem\CleanDir] Cleaned var/cache/modules/Contacts
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Line_Item_Groups
 [Filesystem\CleanDir] Cleaned var/cache/modules/vCals
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Product_Categories
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_PDF_Templates
 [Filesystem\CleanDir] Cleaned var/cache/modules/OAuthTokens
 [Filesystem\CleanDir] Cleaned var/cache/modules/Cases
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Quotes
 [Filesystem\CleanDir] Cleaned var/cache/modules/EAPM
 [Filesystem\CleanDir] Cleaned var/cache/modules/Configurator
 [Filesystem\CleanDir] Cleaned var/cache/modules/FP_Event_Locations
 [Filesystem\CleanDir] Cleaned var/cache/modules/OAuth2Clients
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOD_Index
 [Filesystem\CleanDir] Cleaned var/cache/modules/EmailTemplates
 [Filesystem\CleanDir] Cleaned var/cache/modules/SurveyQuestions
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOS_Products_Quotes
 [Filesystem\CleanDir] Cleaned var/cache/modules/AOR_Charts

```

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by isleshocky77** [comment]
@Dillon-Brown You need anything further on this one?

**Comment by Dillon-Brown** [comment]
@isleshocky77 Looks good to me, just needs a code-review. I'll see if anyone is free, thanks!

File: lib/Robo/Plugin/Commands/CleanCacheCommands.php [hunk 1]
```diff
@@ -2,6 +2,10 @@
```
Comment (code-ph0y) [inline lib/Robo/Plugin/Commands/CleanCacheCommands.php]: Please add copyright statement at the top of the file, before namespace.

File: lib/Robo/Plugin/Commands/CleanCacheCommands.php [hunk 2]
```diff
@@ -14,48 +18,70 @@
...
-                               'cache/diagnostics',
-                               'cache/dynamic_fields',
-                               'cache/feeds',
-                               'cache/import',
-                               'cache/include/javascript',
-                               'cache/jsLanguage',
-                               'cache/pdf',
-                               'cache/themes',
-                               'cache/xml',
+        $cachesToDelete = array('Relationships',
```
Comment (code-ph0y) [inline lib/Robo/Plugin/Commands/CleanCacheCommands.php]: Place 'Relationships', on a newline, For example:
$cachesToDelete = array(
    'Relationships',
    'csv',
    ...
    ...
);

File: lib/Robo/Plugin/Commands/CleanCacheCommands.php (hunk 1 above)
Comment (isleshocky77) [reply lib/Robo/Plugin/Commands/CleanCacheCommands.php]: Added

File: lib/Robo/Plugin/Commands/CleanCacheCommands.php (hunk 2 above)
Comment (isleshocky77) [reply lib/Robo/Plugin/Commands/CleanCacheCommands.php]: Done.

**Pull request merged**